### PR TITLE
Fixing modus select issues

### PR DESCRIPTION
**Pull request by hemantk1325**
## Description

Fixing modus select issue and using native select for select dropdown instead of div.

References #559 , #1334 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: stencil-workspace/src/components/modus-select/modus-select.tsx [hunk 1]
```diff
@@ -46,124 +45,81 @@ export class ModusSelect {
   @Prop() validText: string;
 
   /** (optional) The input value. */
-  @Prop() value: unknown;
+  @Prop({ mutable: true }) value: unknown;
```
Comment (ryan-henness-trimble) [inline stencil-workspace/src/components/modus-select/modus-select.tsx:48]: I know we were mutating the `value` input previously, but I'm not so sure we should be doing that

If we want to change the value of an input internally, I think we should represent that input with internal state. We then could mutate that internal state instead

@msankaran0712 thoughts on this?

File: stencil-workspace/src/components/modus-select/modus-select.scss [hunk 2]
```diff
@@ -27,17 +27,25 @@
...
+    select {
       align-items: center;
       background-color: $modus-input-bg;
       border: solid $rem-1px $modus-input-border-color;
       border-bottom-color: $modus-input-bottom-line-color;
+      color: $modus-select-color;
       display: flex;
+      font-family: $primary-font;
+      font-size: $rem-14px;
+      font-weight: 400;
```
Comment (ryan-henness-trimble) [inline stencil-workspace/src/components/modus-select/modus-select.scss:39]: Is the `font-weight` attribute necessary?

File: stencil-workspace/src/components/modus-select/modus-select.tsx [hunk 3]
```diff
@@ -1,6 +1,5 @@
 // eslint-disable-next-line
-import { Component, Element, Event, EventEmitter, h, Listen, Prop, State } from '@stencil/core';
-import { IconTriangleDown } from '../icons/icon-triangle-down';
+import { Component, Event, EventEmitter, h, Prop } from '@stencil/core';
```
Comment (ryan-henness-trimble) [inline stencil-workspace/src/components/modus-select/modus-select.tsx:2]: This component has a black border when it is interacted with via tabbing in Chrome. When the component is tabbed to, it should have a blue bottom border

It looks like a good example of this working correctly is the Text Input component

File: stencil-workspace/src/components/modus-select/modus-select.tsx (hunk 1 above)
Comment (ryan-henness-trimble) [reply stencil-workspace/src/components/modus-select/modus-select.tsx:48]: We could do this with a `@Watch` function that updates the internal state when the prop updates

File: stencil-workspace/src/components/modus-select/modus-select.tsx [hunk 4]
```diff
@@ -46,124 +45,81 @@ export class ModusSelect {
...
-          <label class="sub-text valid">{this.validText}</label>
-        ) : this.helperText ? (
-          <label class="sub-text helper">{this.helperText}</label>
-        ) : null}
+        <span class="input-container">
+          <select
+            disabled={this.disabled || this.options.length === 0}
+            class={selectClass}
+            aria-label={this.ariaLabel}
+            onChange={(event) => {
```
Comment (ryan-henness-trimble) [inline stencil-workspace/src/components/modus-select/modus-select.tsx:91]: We should add functions for logic within the markup like this

File: stencil-workspace/storybook/stories/components/modus-select/modus-select-storybook-docs.mdx [hunk 5]
```diff
@@ -52,112 +52,21 @@ This component is compatible with Angular reactive forms. This can be achieved t
...
+| `disabled`           | `disabled`             | (optional) Whether the input is disabled.            | `boolean`   | `undefined` |
+| `errorText`          | `error-text`           | (optional) The input's error text.                   | `string`    | `undefined` |
+| `helperText`         | `helper-text`          | (optional) The input's helper text.                  | `string`    | `undefined` |
+| `label`              | `label`                | (optional) The input label.                          | `string`    | `undefined` |
+| `options`            | --                     | The options for the dropdown list.                   | `unknown[]` | `[]`        |
+| `optionsDisplayProp` | `options-display-prop` | The options property to render in the dropdown list. | `string`    | `undefined` |
+| `required`           | `required`             | (optional) Whether the input is required.            | `boolean`   | `undefined` |
+| `size`               | `size`                 | (optional) The input's size.                         | `"large"    | "medium"`   |
```
Comment (sankarNous) [inline stencil-workspace/storybook/stories/components/modus-select/modus-select-storybook-docs.mdx:67]: is 'Large' datatype?

File: stencil-workspace/src/components/modus-select/modus-select.tsx [hunk 6]
```diff
@@ -48,122 +49,108 @@ export class ModusSelect {
...
-          <label class="sub-text error">{this.errorText}</label>
-        ) : this.validText ? (
-          <label class="sub-text valid">{this.validText}</label>
-        ) : this.helperText ? (
-          <label class="sub-text helper">{this.helperText}</label>
-        ) : null}
+      ) : null;
+
+    const optionList =
+      this.options.length > 0 &&
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-select/modus-select.tsx:118]: `this.options?.`

File: stencil-workspace/src/components/modus-select/modus-select.tsx [hunk 7]
```diff
@@ -48,122 +49,108 @@ export class ModusSelect {
...
+    return (
+      <div
+        aria-disabled={this.disabled ? 'true' : undefined}
+        aria-label={this.ariaLabel}
+        aria-required={this.required}
+        class={this.disabled ? 'disabled' : undefined}>
+        {labelContainer}
+        <span class="input-container">
+          <select
+            disabled={this.disabled || this.options.length === 0}
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-select/modus-select.tsx:142]: why to disable it when no options?

File: stencil-workspace/src/components/modus-select/modus-select.scss [hunk 8]
```diff
@@ -27,17 +27,24 @@
...
-    button {
+    select {
       align-items: center;
       background-color: $modus-input-bg;
       border: solid $rem-1px $modus-input-border-color;
       border-bottom-color: $modus-input-bottom-line-color;
+      color: $modus-select-color;
       display: flex;
+      font-family: $primary-font;
+      font-size: $rem-14px;
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-select/modus-select.scss:38]: font size should be $rem-12px. Refer to https://modus.trimble.com/components/web/inputs/styles/ and please verify for font size for label, select text and placeholder text for 'medium'(default) and 'large' sizes

File: stencil-workspace/storybook/stories/components/modus-select/modus-select-storybook-docs.mdx [hunk 9]
```diff
@@ -52,112 +52,21 @@ This component is compatible with Angular reactive forms. This can be achieved t
 <modus-select label="Select Demo 4" valid-text="Valid demo"></modus-select>
```
Comment (msankaran0712) [inline stencil-workspace/storybook/stories/components/modus-select/modus-select-storybook-docs.mdx:52]: Please add one example for large size Select

File: stencil-workspace/src/components/modus-select/modus-select.scss [hunk 10]
```diff
@@ -27,17 +27,24 @@
...
       height: $btn_height;
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-select/modus-select.scss:39]: where is $btn_height defined ?

File: stencil-workspace/src/components/modus-select/modus-select.scss [hunk 11]
```diff
@@ -136,33 +87,22 @@
...
-      button {
-        background-color: $modus-input-disabled-bg;
-        border: $rem-1px solid $modus-input-disabled-border-color;
-        border-bottom-color: $modus-input-disabled-bottom-line-color;
-
-        .dropdown-text {
-          color: $modus-input-disabled-color;
-        }
-      }
+select:focus {
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-select/modus-select.scss:92]: On focus, Select input should only have a bottom blue border, no full border, refer https://modus.trimble.com/components/web/inputs/styles/#behaviors

File: stencil-workspace/src/components/modus-select/modus-select.scss [hunk 12]
```diff
@@ -136,33 +87,22 @@
...
+  border-color: $modus-select-focus-color;
+  box-shadow: 0 0 0 2px $modus-select-focus-color;
+  outline: none;
+}
 
-      svg path {
-        fill: $modus-input-disabled-color;
-      }
+select.error:focus {
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-select/modus-select.scss:98]: no need for a full border for error and valid states too. Please remove this CSS block.

File: stencil-workspace/src/components/modus-select/modus-select.scss [hunk 13]
```diff
@@ -136,33 +87,22 @@
       color: $modus-input-validation-success-color;
     }
   }
+}
```
Comment (msankaran0712) [inline stencil-workspace/src/components/modus-select/modus-select.scss:90]: `.sub-text` font size should be $rem-12px

**Review by msankaran0712 — CHANGES_REQUESTED** [review]

File: stencil-workspace/storybook/stories/components/modus-select/modus-select-storybook-docs.mdx (hunk 5 above)
Comment (msankaran0712) [reply stencil-workspace/storybook/stories/components/modus-select/modus-select-storybook-docs.mdx:67]: @sankarNous `large` and `medium` are options for `size` property

Comment (msankaran0712) [reply stencil-workspace/storybook/stories/components/modus-select/modus-select-storybook-docs.mdx:67]: @hemantk1325 please copy the properties section from README again and paste it into the storybook

Expected:
image

File: stencil-workspace/src/components/modus-select/modus-select.e2e.ts [hunk 14]
```diff
@@ -1,6 +1,6 @@
 import { newE2EPage } from '@stencil/core/testing';
 
-describe('modus-select', () => {
+xdescribe('modus-select', () => {
```
[the remark this answers is not in the document]
Comment (hemantk1325) [reply stencil-workspace/src/components/modus-select/modus-select.e2e.ts]: added test cases

File: stencil-workspace/src/components/modus-select/modus-select.scss (hunk 13 above)
Comment (hemantk1325) [reply stencil-workspace/src/components/modus-select/modus-select.scss:94]: done

File: stencil-workspace/src/components/modus-select/modus-select.scss (hunk 12 above)
Comment (hemantk1325) [reply stencil-workspace/src/components/modus-select/modus-select.scss]: made the changes

File: stencil-workspace/src/components/modus-select/modus-select.scss (hunk 11 above)
Comment (hemantk1325) [reply stencil-workspace/src/components/modus-select/modus-select.scss]: done

File: stencil-workspace/src/components/modus-select/modus-select.scss (hunk 10 above)
Comment (hemantk1325) [reply stencil-workspace/src/components/modus-select/modus-select.scss]: done

File: stencil-workspace/storybook/stories/components/modus-select/modus-select-storybook-docs.mdx (hunk 9 above)
Comment (hemantk1325) [reply stencil-workspace/storybook/stories/components/modus-select/modus-select-storybook-docs.mdx:52]: Added example for large size

File: stencil-workspace/src/components/modus-select/modus-select.tsx (hunk 6 above)
Comment (hemantk1325) [reply stencil-workspace/src/components/modus-select/modus-select.tsx]: done

File: stencil-workspace/src/components/modus-select/modus-select.tsx [hunk 15]
```diff
@@ -48,122 +49,108 @@ export class ModusSelect {
...
+    this.handleItemSelect(selectedValue);
   }
 
-  showDropdown(): void {
-    this.visible = true;
-
-    const activeOptionIndex = this.options?.findIndex((option) => option === this.value);
-    if (activeOptionIndex > -1) {
-      this.activeItemIndex = activeOptionIndex;
+  renderText(): JSX.Element | null {
```
[the remark this answers is not in the document]
Comment (hemantk1325) [reply stencil-workspace/src/components/modus-select/modus-select.tsx]: done

File: stencil-workspace/src/components/modus-select/modus-select.tsx [hunk 16]
```diff
@@ -48,122 +49,108 @@ export class ModusSelect {
...
+        ? 'error'
+        : this.validText
+        ? 'valid'
+        : this.disabled
+        ? 'disabled'
+        : ''
+    }`;
+    let labelText: JSX.Element | null = null;
+
+    if (this.errorText) {
```
[the remark this answers is not in the document]
Comment (hemantk1325) [reply stencil-workspace/src/components/modus-select/modus-select.tsx]: done

File: stencil-workspace/src/components/modus-select/modus-select.tsx [hunk 17]
```diff
@@ -1,6 +1,13 @@
 // eslint-disable-next-line
```
[the remark this answers is not in the document]
Comment (hemantk1325) [reply stencil-workspace/src/components/modus-select/modus-select.tsx]: done

File: stencil-workspace/src/components/modus-select/modus-select.tsx [hunk 18]
```diff
@@ -1,6 +1,13 @@
...
-import { Component, Element, Event, EventEmitter, h, Listen, Prop, State } from '@stencil/core';
-import { IconTriangleDown } from '../icons/icon-triangle-down';
+import {
+  Component,
+  Event,
+  EventEmitter,
+  h,
```
[the remark this answers is not in the document]
Comment (hemantk1325) [reply stencil-workspace/src/components/modus-select/modus-select.tsx:6]: done

File: stencil-workspace/storybook/stories/components/modus-select/modus-select-storybook-docs.mdx (hunk 5 above)
Comment (hemantk1325) [reply stencil-workspace/storybook/stories/components/modus-select/modus-select-storybook-docs.mdx]: done

File: stencil-workspace/storybook/stories/components/modus-select/modus-select.stories.tsx [hunk 19]
```diff
@@ -23,15 +23,29 @@ const Template = () => html`
...
+  <modus-select
+    error-text="Error demo"
+    aria-label="select"
+    label="Select Demo 3"></modus-select>
+  <modus-select
+    label="Select Demo 4"
+    aria-label="select"
+    valid-text="Valid demo"></modus-select>
+    <modus-select
+    id="select-demo-5"
```
Comment (msankaran0712) [inline stencil-workspace/storybook/stories/components/modus-select/modus-select.stories.tsx:44]: Please give the label as "Large Select"

**Review by msankaran0712 — CHANGES_REQUESTED** [review]
Also, Increase the container height in storybook 
image

File: stencil-workspace/src/components/modus-select/modus-select.tsx (hunk 1 above)
Comment (msankaran0712) [reply stencil-workspace/src/components/modus-select/modus-select.tsx:49]: @ryan-henness-trimble it looks like in `modus-text-input`  we are mutating the `value` inside the component without marking it as `mutable`

File: stencil-workspace/src/components/modus-select/modus-select.tsx (hunk 6 above)
Comment (hemantk1325) [reply stencil-workspace/src/components/modus-select/modus-select.tsx]: done

File: stencil-workspace/src/components/modus-select/modus-select.tsx (hunk 7 above)
Comment (hemantk1325) [reply stencil-workspace/src/components/modus-select/modus-select.tsx]: done

File: stencil-workspace/src/components/modus-select/modus-select.scss (hunk 8 above)
Comment (hemantk1325) [reply stencil-workspace/src/components/modus-select/modus-select.scss]: done

File: stencil-workspace/storybook/stories/components/modus-select/modus-select.stories.tsx (hunk 19 above)
Comment (hemantk1325) [reply stencil-workspace/storybook/stories/components/modus-select/modus-select.stories.tsx:44]: done

**Comment by msankaran0712** [comment]
Closing the PR due to branching issues

**Comment by msankaran0712** [comment]
Created a new PR for the issue https://github.com/trimble-oss/modus-web-components/pull/1407

File: stencil-workspace/src/components/modus-select/modus-select.tsx (hunk 1 above)
Comment (msankaran0712) [reply stencil-workspace/src/components/modus-select/modus-select.tsx:49]: @ryan-henness-trimble We have close this PR due to some branch issues and please check the new PR https://github.com/trimble-oss/modus-web-components/pull/1407

**Pull request closed without merging**